src/index.js

```diff
@@ -37,48 +36,61 @@ const BUTTONS_ARRAY = [
     },
     {
         value: "*",
-        characters: ""
     },
     {
         value: 0,
         characters: "+"
     },
     {
         value: "#",
-        characters: ""
     },
     {
         value: "Call",
-        characters: ""
     },
     {
         value: "X",
-        characters: ""
     }
 ]
 
     ; (() => {
-
         // HEADER
         const headerElement = document.querySelector("header")
         const addNumberElement = document.querySelector("p")
         addNumberElement.innerText = "Add Number"
 
+        const handleDisplayElement = () => {
+            const displayElement = document.getElementById("display")
+            console.log(displayElement)
+        }
+
         // MAIN
+
         const mainElement = document.querySelector("main")
+        const numbersToDisplay = []
 
         BUTTONS_ARRAY.forEach((button, index) => {
+            const handleDialButtonOnClick = () => {
+                numbersToDisplay.push(buttonValueElement.innerHTML)
+                console.log(numbersToDisplay)
+            }
             const dialButtonElement = document.createElement("div")
             dialButtonElement.className = "dial-button"
+            dialButtonElement.onclick = handleDialButtonOnClick
 
             const buttonValueElement = document.createElement('h2')
             buttonValueElement.innerHTML = button.value
             dialButtonElement.appendChild(buttonValueElement)
 
             const buttonCharactersElement = document.createElement('p')
             buttonCharactersElement.innerHTML = button.characters
-            dialButtonElement.appendChild(buttonCharactersElement)
+            if (button.characters !== undefined) {
+                dialButtonElement.appendChild(buttonCharactersElement)
+            }
 
             mainElement.appendChild(dialButtonElement)
+
         })
+        handleDisplayElement()
```

Comment:
If you didn't call this function here. Would it change anything? I don't think so as initially display should be empty.